streams/src/main/java/org/apache/kafka/streams/kstream/Joined.java

```diff
@@ -132,6 +136,19 @@ private Joined(final Serde<K> keySerde,
         return this;
     }
 
+    /**
+     * Configure the retention time for events in the join.
+     * Events are only considered for joining while they are retained.
```

Comment:
Actually, this is making me realize something.

We moved "retention" to Materialized because it's only important for visibility in IQ now. But join windows aren't visible at all in IQ, therefore we don't need the concept of retention for Join windows. It is sufficient to set join window retention equal to the grace period.

Reply:
Sounds good.